diff --git a/deflate.c b/deflate.c
index 630cce2..19d2c1a 100644
--- a/deflate.c
+++ b/deflate.c
@@ -69,7 +69,7 @@ const char PREFIX(deflate_copyright)[] = " deflate 1.3.1 Copyright 1995-2024 Jea
 /* ===========================================================================
  *  Architecture-specific hooks.
  */
-#ifdef S390_DFLTCC_DEFLATE
+#if 1
 #  include "arch/s390/dfltcc_deflate.h"
 /* DFLTCC instructions require window to be page-aligned */
 #  define PAD_WINDOW            PAD_4096
